config.template.toml

```diff
@@ -165,11 +165,119 @@ model = "gpt-4o"
 
 [llm.gpt4o-mini]
+# Model to use
+model = "gpt-4o-mini"
 # API key to use
 api_key = "your-api-key"
 
-# Model to use
-model = "gpt-4o-mini"
+
+################################### Router ###################################
+# litellm Router configuration
+# https://litellm.vercel.app/docs/routing
+##############################################################################
+
+# The "router_config"section can only occur once in the config.toml file!
+#[router_config]
+
+# The default model to use for routing, which must be part of the "models" entries below.
+#default_model = "gpt-4o"
+
+# The strategy to use for routing (e.g., usage-based-routing, cost-based-routing)
+#routing_strategy = "usage-based-routing"
+
+# Number of retries to attempt when an operation fails
+#num_retries = 3
+
+# waits min. 5s before retrying request
+#retry_after = 15
+
+# cooldown model if it fails > 1 call in a minute
+#allowed_fails = 1
+
+# cooldown the deployment for 100 seconds if it num_fails > allowed_fails
+#cooldown_time = 100
+
+# raise timeout error if call takes > 240s
+#timeout = 240
+
+# Whether to cache responses for reuse
+#cache_responses = false
+
+# Additional cache configuration options, see documentation for details
+#[router_config.cache_kwargs]
+
+#[router_config.retry_policy]
+#exceptions_to_retry = [ "RateLimitError",]  # List of exceptions to retry on
+#max_retries = 8  # Maximum number of retries for the specified exceptions
+#retry_after = 15  # Time to wait before the first retry for the specified exceptions
+#retry_after_multiplier = 2  # Multiplier for exponential backoff calculation
+
+# --- Important Notes ---
+# - The model entries below are just examples, you can add, edit or remove these as needed.
+# - For each model entry, there is a corresponding set of "litellm_params" section,
+#   which must be specified in the exact same syntax as shown below.
+# - The syntax with the brackets [[ ]] is mandatory, even if there is only one model entry,
+#   or the whole config file may not be loaded!
+# - The model name must be unique in the list!
+# - The model name can be the same as the model alias.
+# - The empty lines are only added for readability, they are not required.
+# ----
+
+
+# --- 1st model example: OpenRouter w/ Qwen 2.5
+
+# See: https://openrouter.ai/models/
+
+#[[router_config.models]]
+#model_name = "or-qwen-2.5"
+
+#[router_config.models.litellm_params]
+#model = "openrouter/qwen/qwen-2.5-72b-instruct"
+#api_base = "https://openrouter.ai/api/v1" # Optional with litellm if model starts with "openrouter/"
+#api_key = "sk-or-v1-xxx"
+# Note: all other models may have these parameters, which are optional,
+# but are only mentioned here for brevity. These are example values!
+#max_retries = 3       # Number of retries to attempt when an operation fails
+#max_tokens = 4000     # Maximum number of output tokens
+#tpm = 10000           # Tokens per minute limit
+#rpm = 6               # Requests per minute limit
```

Comment:
Not a review of sorts, just thinking out loud: so it looks like this is what we have to do, put the responsibility on the user to find out the limits of their llm/provider, and set them per model? 🤔

Reply:
Those params are optional, and I'm not sure what litellm does internally if they're left out here.

Reply:
Right, but what I mean is:
- remember that Groq issue with terrible token limits? We don't have configuration implemented with limits per tokens, this does.
- it doesn't do it automatically, same for rpm, it doesn't read / know the limits per provider, we need to set them.

The first is an improvement, even if it doesn't do the second.